src/MPS.jl

```diff
@@ -612,7 +612,7 @@ function evolve!(::Canonical, ψ::AbstractAnsatz, mpo::AbstractMPO; threshold, m
     evolve!(NonCanonical(), ψ, mpo; threshold=nothing, maxdim=nothing, normalize=false, kwargs...) # set maxdim and threshold to nothing so we truncate from Canonical form
```

Comment:
I just noticed this can give us problems, since the `threshold=nothing`, `maxdim=nothing` and `normalize=false` will be overriden by `kwargs...`. I believe we must put the `kwargs...` before.

Also would you mind moving the comment to the line above? can be problematic to read in the text editor if you don't wrap the lines.

```suggestion
    # set `maxdim` and `threshold` to `nothing` so we later truncate in the `Canonical` form
    evolve!(NonCanonical(), ψ, mpo; kwargs..., threshold=nothing, maxdim=nothing, normalize=false)
```